src/uuid.cr

```diff
@@ -161,9 +161,9 @@ struct UUID
     end
   end
 
-  # Returns 16-byte slice.
-  def to_slice
```

Comment:
What about adding a deprecation annotation? 
But still fix #7894 by doing `@bytes.to_slice.dup` until we remove the method completely